src/price_data.py

```diff
@@ -267,33 +269,41 @@ def _get_price_bitpanda_pro(
         # if there were no trades in the requested time frame, the
         # returned data will be empty
         for t in timeframes:
-            end = utc_time
-            begin = utc_time - datetime.timedelta(minutes=t)
-
-            # https://github.com/python/mypy/issues/3176
-            params: dict[str, Union[int, str]] = {
-                "unit": "MINUTES",
-                "period": t,
-                "from": begin.isoformat(),
-                "to": end.isoformat(),
-            }
-            r = requests.get(baseurl, params=params)
-
-            assert r.status_code == 200
+            # if no trades can be found, move 30 min window to the past
+            for o in range(0, 300, 30):
+                end = utc_time.astimezone(pytz.UTC) - datetime.timedelta(minutes=o)
```

Comment:
I think, that we do not need pytz here. Please have a look at [this stackoverflow answer](https://stackoverflow.com/a/2175170/8979290).